Mathlib/Probability/Variance.lean

```diff
@@ -341,4 +343,60 @@ theorem IndepFun.variance_sum [IsProbabilityMeasure μ] {ι : Type*} {X : ι →
       rw [IH (fun i hi => hs i (mem_insert_of_mem hi))
           (h.mono (by simp only [coe_insert, Set.subset_insert]))]
 
+theorem abs_le_max_abs_abs_ae {a b : ℝ} {X : Ω → ℝ}
+    (ha : ∀ᵐ (ω : Ω) ∂μ, a ≤ X ω) (hb : ∀ᵐ (ω : Ω) ∂μ, X ω ≤ b) :
+    ∀ᵐ (ω : Ω) ∂μ, |X ω| ≤ max |a| |b| := by
+  filter_upwards [ha, hb] with ω using abs_le_max_abs_abs
+
+theorem memℒp_of_bounded [IsFiniteMeasure μ]
+    {a b : ℝ} {X : Ω → ℝ} (ha : ∀ᵐ (ω : Ω) ∂μ, a ≤ X ω) (hb : ∀ᵐ (ω : Ω) ∂μ, X ω ≤ b)
+    (hX : AEMeasurable X μ) (p : ENNReal) : Memℒp X p μ :=
+  let c := max |a| |b|
+  (memℒp_const c).mono' hX.aestronglyMeasurable (abs_le_max_abs_abs_ae ha hb)
+
+/-! ### Popvinciu's inequality on variances -/
+/--Variance is bounded by `((b - a) / 2) ^ 2`
+if almost everywhere real-valued measurable function `X` satisfies `a ≤ X ≤ b` almost everywhere.-/
+
+lemma variance_square_bounded [IsProbabilityMeasure μ] {a b : ℝ} {X : Ω → ℝ}
+    (ha : ∀ᵐ ω ∂μ, a ≤ X ω) (hb : ∀ᵐ ω ∂μ, X ω ≤ b) (hX : AEMeasurable X μ) :
+    variance X μ ≤ ((b - a) / 2) ^ 2 :=
+  let c := max |a| |b|
+  have hX_int₁ : Integrable (fun ω ↦ -X ω ^ 2) μ :=
+    ((memℒp_of_bounded ha hb hX 2).integrable_sq).neg
+  have hX_int₂ : Integrable (fun ω ↦ (a + b) * X ω) μ :=
+    ((integrable_const c).mono' hX.aestronglyMeasurable
+      (abs_le_max_abs_abs_ae ha hb)).const_mul (a + b)
+  have h0 : 0 ≤ - ∫ (ω : Ω), (X ω ^ 2) ∂μ + (a + b) * ∫ (ω : Ω), (X ω) ∂μ - a * b :=
+    calc
+      _ ≤ ∫ (ω : Ω), (b - X ω) * (X ω - a) ∂μ := by
+        apply integral_nonneg_of_ae
+        filter_upwards [ha, hb] with ω ha' hb'
+        exact mul_nonneg (by linarith : 0 ≤ b - X ω) (by linarith : 0 ≤ X ω - a)
+      _ = ∫ (ω : Ω), - X ω ^ 2 + (a + b) * X ω - (a * b) ∂μ :=
+        integral_congr_ae <| ae_of_all μ fun ω ↦ by ring
+      _ = ∫ (ω : Ω), - X ω ^ 2 + (a + b) * X ω ∂μ - ∫ (_ : Ω), (a * b) ∂μ :=
```

Comment:
here is a second example of parentheses and types that can be removed. Please make similar changes elsewhere.
```suggestion
      _ = ∫ ω, - X ω ^ 2 + (a + b) * X ω ∂μ - ∫ _, a * b ∂μ :=
```

Reply:
Thank you for your suggestion about parentheses. I'll revise it.